src/array/fixed_size_list.rs

```diff
@@ -246,6 +246,94 @@ where
     }
 }
 
+/// An iterator over `N` elements of the iterator at a time.
+pub struct FixedSizeArrayChunks<const N: usize, I: IntoIterator> {
+    /// An owned iterator
+    iter: I::IntoIter,
+}
+
+impl<const N: usize, I: IntoIterator> FixedSizeArrayChunks<N, I> {
+    /// Returns a new [`FixedSizeArrayChunks`]
+    fn new(iter: I::IntoIter) -> Self {
+        Self { iter }
+    }
+}
+
+impl<const N: usize, I: IntoIterator> Iterator for FixedSizeArrayChunks<N, I> {
+    type Item = [I::Item; N];
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut data: [MaybeUninit<I::Item>; N] =
+        // Safety:
+        // - https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element
+         unsafe { MaybeUninit::uninit().assume_init() };
+
+        let mut total_elements_written: usize = 0;
+        self.iter
+            .by_ref()
+            .take(N)
+            .enumerate()
+            .for_each(|(array_index, val)| {
+                data[array_index].write(val);
+                total_elements_written += 1;
+            });
+
+        (total_elements_written == N).then(|| {
```

Comment:
If the number of elements remaining in the iterator were less than `N`, and we return `None` due to this condition, is it safe to have values written to some `MaybeUninit` elements of the array, but never initialized?

Reply:
> the written elements leak

Would this mitigate the issue?
```rust
if total_elements_written == N {
    Some(data.map(|elem| {
        // Safety:
        // - We only initialize if we acually wrote to this element.
        unsafe { elem.assume_init() }
    }))
} else {
    // For each elem in the array, drop if we wrote to it to prevent memory leaks.
    for elem in &mut data[0..total_elements_written] {
        // Safety:
        // - This element was initialized as indicated by `total_elements_written`.
        unsafe {
            // Drop the value to prevent a memory leak.
            elem.assume_init_drop();
        }
    }
    None
}
```